### DISCUSSION TITLE
`nx affected` exit code when nothing has changed?

### DISCUSSION THREAD

**Discussion by leonardochaia** [nrwl/nx#6423] [Feature Requests]
Hey nx Folks!

We've been using nx for over a year in our little angular monorepo. I'm currently migrating our CI builds and was hoping to speed things up a bit. I have a question, if the answer is no then it would be a feature request :+1: 

I'm trying to run certain steps only when something was lint/built/tested.

I'm basically looking for something like this:
```csharp
nx affected --error-when-changes # .. other args
if [ "$?" = "166" ]; then
   # .. set a pipeline variable HAS_CHANGES=false
   exit 0 #end this step earlier and successfully
fi
```
`--error-when-changes` could be a flag that makes the CLI exit with a custom exit code like `166`.

Is there any other way (other than `grep`ing stdout) to know when there has been any changed/affected project?

**Comment by CaffeinatedCodeMonkey** [comment]
I had to do something similar to this. But I found an undocumented flag that helped with it: `--plain`. It spits out a very simple string of changes, instead of the whole display text thing. I imagine you could do a `-z` on the results to see if nothing's changed.
